src/main/java/org/stellar/sdk/StrKey.java

```diff
@@ -22,6 +22,15 @@
         public int getValue() {
             return value;
         }
+
+        public static VersionByte findByValue(byte value) {
+            for (VersionByte versionByte : values()) {
+                if (value == versionByte.value) {
+                    return versionByte;
+                }
+            }
+            throw new IllegalArgumentException("No matching versionByte found.");
```

Comment:
Should we throw a custom exception here?

Reply:
Yeah probably. If we were supporting Java 8+ we could return an Optional but I think we might be supporting older versions of Java. @tamirms do you know?

Reply:
we're limited to using Java 6 because most android devices are stuck on that version. However, you can use com.google.common.base.Optional if you prefer . We use it in other parts of the sdk

Reply:
Thank you, I think it's a good idea to use `com.google.common.base.Optional` when we can't upgrade to Java 8+.